sklearn/utils/estimator_checks.py

```diff
@@ -304,7 +304,7 @@ def check_estimator(Estimator):
         except SkipTest as message:
             # the only SkipTest thrown currently results from not
             # being able to import pandas.
-            warnings.warn(message, SkipTestWarning)
+            warnings.warn(str(message), SkipTestWarning)
```

Comment:
For the record I bumped into the same problem. This is clearly a bug in our code, trying to provide an exception object into `warnings.warn` only work if none of the warnings filters use a regex. Here is a snippet to show what I means:

```py
import warnings

exc = ValueError()
warnings.resetwarnings()
warnings.warn(exc)

warnings.filterwarnings('always', message='regex_goes_here')
warnings.warn(exc)  # exception here TypeError: expected string or bytes-like object
```

Reply:
While you are at it, you may as well rename `message` -> `exception` to improve readability.

Reply:
where was the log? is it `./build_tools/circle/build_doc.sh`?